pkg/types/vector.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"encoding/binary"
+	"strconv"
+	"unsafe"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/pingcap/errors"
+)
+
+func init() {
+	var buf [4]byte
+	binary.NativeEndian.PutUint32(buf[:], 0x2)
+	if buf[0] == 0x02 && buf[1] == 0x00 && buf[2] == 0x00 && buf[3] == 0x00 {
+		return
+	}
+	panic("VectorFloat32 only supports little endian")
```

Comment:
Does it mean that TiDB cannot bootstrap even if the user does not the use vector feature?

Reply:
Yes. This is a protection. Common hardwares like x86, x64 and AWS Graviton are all little endian machines. Other TiDB features may also break in BigEndian as well, considering that memory direct copy is used in so many places.